### DISCUSSION TITLE
Crossposts are not combined

### DISCUSSION THREAD

**Discussion by kroese** [aeharding/voyager#465] [Ideas]
When a post is posted to multiple communities, it will show up multiple times in the feed (one time for each community).

It was better if those kind of posts were combined into a single post referencing multiple communities (like the official Lemmy UI does).

**Comment by aeharding** [comment]
If anyone could track down the code where lemmy-ui does this, that would be helpful

**Comment by mdecourcy** [comment]
@aeharding https://github.com/LemmyNet/lemmy-js-client/blob/2b757d39ec3e34fff952cc6ae2b61883f34b7d7e/src/types/GetPostResponse.ts#L6

looks like cross post list is in GetPostResponse.

Can't find combining of posts. I know that lemmy ui lists where it's cross posted.

**Comment by aeharding** [comment]
It might just list crossposts, I'd be surprised if lemmy-ui is deduplicating in the feed

**Comment by mdecourcy** [comment]
Yeah, after more digging it just seems to list them. Would need to be a server side function as well, otherwise comments would get duplicated or only show up on one post.

**Comment by aeharding** [comment]
I'm going to convert this to a discussion because this is a good topic worth discussion, but at the moment there's no actionable work for Voyager

**Comment by kroese** [comment]
Im almost sure that it de-duplicates, because I never saw duplicates when using lemmy-ui. It became instantly noticeable as soon as I tried Voyager.

**Comment by mdecourcy** [comment]
Lol both of the servers I have accounts on seem to be unreachable, going to create an account on a smaller instance to test.

**Comment by mdecourcy** [comment]
<img width="557" alt="image" src="https://github.com/aeharding/voyager/assets/49794076/0e0126dd-37aa-4d86-b8b3-09bf61d66ebe">

1. Created a new post.
2. Clicked cross post button after posting.
3. Cross posted to other test communities.
4. Subscribed to the test communities, then viewed my “subscribed” feed.

Seems like lemmy-ui doesn't dedup.

Would be great if this was a thing, though. Could be worth opening an issue in lemmyNet/lemmy

**Comment by kroese** [comment]
@mdecourcy But those dont show the links to the other communities, so it seems they are not recognized as crossposts somehow.

It is only the posts that are recognized as crossposts (the ones that mention the other communities) that I dont see duplicates.

**Reply by mdecourcy** [reply]
Could be due to both lemmy.ml and lemmy.world slowness. I'll xpost to smaller servers.

**Reply by mdecourcy** [reply]
Still not getting the cross post text, unsure what I'm doing wrong. Maybe there needs to be a link/picture for it to be picked up?

**Reply by kroese** [reply]
You cannot crosspost between servers I guess? How I reproduce it is by posting the same url/title/body to multiple communities on the same server. Then Lemmy automaticly detects them as crossposts (even though they were created seperately).

Edit: I can even crosspost to the same community this way. So just repeat the same post to the same group a couple of times, and then check the community feed.

**Comment by mdecourcy** [comment]
Yep, needed to add a link. 

Screenshot of lemmy-ui:
<img width="645" alt="image" src="https://github.com/aeharding/voyager/assets/49794076/fc752eba-c91c-4b18-9f46-3d2432e15c17">

Screenshot from vger.app:
<img width="729" alt="image" src="https://github.com/aeharding/voyager/assets/49794076/1d3b7ccf-bb62-4322-bb6c-1569163ebd4b">


Findings: Does seem like posts are deduped. Comments aren't shared between the cross posts, but the link only shows up once on the feed.

**Reply by mdecourcy** [reply]
@aeharding I can dig into this a bit later tonight, but I have some work I need to wrap up

**Comment by kroese** [comment]
I think it just chooses the oldest post (which is also likely to have the most comments) when picking which one to show.

**Reply by mdecourcy** [reply]
Yes, the original post seems to be shown in the feed.

**Comment by mdecourcy** [comment]
https://github.com/LemmyNet/lemmy-ui/blob/9c489680de48247cf85b1a4b33f3d0d1a8f88673/src/shared/components/post/post-listings.tsx#L116

Can't believe I didn't see this earlier 🤦‍♂️ going to work on a feature branch now.

**Comment by mdecourcy** [comment]
I haven't been able to get any work done on this as work has been busy, if anyone wants to pick this up, feel free to!
